comp/autoscaling/datadogclient/mock/mock.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build test
+
+package mock
+
+import (
+	"sync"
+
+	"gopkg.in/zorkian/go-datadog-api.v2"
+
+	datadogclient "github.com/DataDog/datadog-agent/comp/autoscaling/datadogclient/def"
+)
+
+var _ datadogclient.Component = (*mockDatadogClient)(nil)
+
+// Provides is a mock for component output
+type Provides struct {
+	Comp Component
+}
+
+// NewMock returns a mock for datadogclient component.
+func NewMock() Provides {
```

Comment:
Constructor for mocks should take a `t *testing.T` as input, event if they don't use it (https://datadoghq.dev/datadog-agent/components/creating-components/#the-mock-folder). This allows future improvement on the mock to use `testing.T` without changing the mock constructor call across the code base and align mock usage for all components.

Also, since we're in the `mock` package we can just call this funciton `New`.